src/optimizer/remove_unused_columns.cpp

```diff
@@ -386,43 +403,50 @@ bool BaseColumnPruner::HandleStructExtractRecursive(Expression &expr, optional_p
 	if (!function.bind_info) {
 		return false;
 	}
-	if (function.children[0]->return_type.id() != LogicalTypeId::STRUCT) {
+	auto &child = function.children[0];
+	if (child->return_type.id() != LogicalTypeId::STRUCT) {
 		return false;
 	}
 	auto &bind_data = function.bind_info->Cast<StructExtractBindData>();
 	indexes.push_back(bind_data.index);
 	// struct extract, check if left child is a bound column ref
-	if (function.children[0]->GetExpressionClass() == ExpressionClass::BOUND_COLUMN_REF) {
+	if (child->GetExpressionClass() == ExpressionClass::BOUND_COLUMN_REF) {
 		// column reference - check if it is a struct
-		auto &ref = function.children[0]->Cast<BoundColumnRefExpression>();
+		auto &ref = child->Cast<BoundColumnRefExpression>();
 		if (ref.return_type.id() != LogicalTypeId::STRUCT) {
 			return false;
 		}
-		colref = &ref;
+		colref = unique_ptr_cast<Expression, BoundColumnRefExpression>(std::move(child));
```

Comment:
This should be conditional, based on the question: does the upstream scan support this?